File: 수학/leetcode118(Easy)-Pascal's Triangle.js
// 60ms(87.54%), 42.2MB
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function (numRows) {
  const result = []

  for (let line = 0; line < numRows; line++) {
    const row = new Array(line + 1).fill(1)
    if (line >= 2) {
      for (let i = 1; i < line; i++) {
        row[i] = result[line - 1][i - 1] + result[line - 1][i]
      }
    }
    result.push(row)
  }

  return result
}

File: 수학/leetcode118(Easy)-Pascal's Triangle.py
# 23ms(98.93%), 14MB
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = []
        
        for line in range(numRows):
            row = [1] * (line + 1)
            if line >= 2:
                for i in range(1, line):
                    row[i] = result[line - 1][i - 1] + result[line - 1][i]
            result.append(row)
            
        return result